### ISSUE TITLE
FilterDate metadata manipulator triggers error

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/mik#128]
Running the FilterDate manipulator results in this error:

```
Missing argument 2 for mik\\metadatamanipulators\\MetadataManipulator::__construct(),
```

Argument 2 is $input, should be $paramsArray.

**Comment by mjordan** [comment]
In ca53886ba252290d3fd88ea7d70e8a5a55f56485 I've rewritten this manipulator so that the cached metadata field name and the XPath to the target element are parameterized in the .ini file like this:

```
metadatamanipulators[] = "FilterDate|Date|/originInfo/dateCreated"
```

But, there is still a lot of specific logic in the manipulator required to modify the value of the snippet that it might not be worth parameterizing at all. Or, we could make this  manipulator generic enough that it would meet a few common cases and if people needed additional functionality, they'd have to write their own or add to this one. I will continue to see if I can generalize this manipulator a bit more however.

Also, @MarcusBarnes I'd like to propose that we rename this manipulator from FilterDate to NormalizeDate, since its more descriptive of what we're doing. Is that OK with you?

**Comment by mjordan** [comment]
OK, I think this manipulator can be pretty general now, but only if we use it to normalize dates that are destined for MODS' originInfo/dateIssued, dateCreated, dateCaptured, dateValid, dateModified, copyrightDate, and dateOther child elements. This should pretty much cover most dates used in MODS. Also, the entry in the .ini file can be much simplified, e.g.,

```
metadatamanipulators[] = "FilterDate|Date|dateCreated"
```
where `Date` is the date field in the cached (raw) metadata and `dateCreated` is the target child element of MODS' `<originInfo>` wrapper.

I'll push this now but test over the next day or two with some patterns I find in the CONTENTdm metadata.

**Comment by mjordan** [comment]
OK, this is working out nicely. The manipulator now checks for incoming dates of format \d\d-\d\d-\d\d\d\d, \d\d\d\d \d\d \d\d, and empty/nonstring. Here are two sample configs:

CSV/Cartoons:
```
metadatamanipulators[] = "FilterDate|Date|dateCreated"
```
Cdm/Vanpunk:
```
metadatamanipulators[] = "FilterDate|date|dateIssued"
```
If the specified input date field is empty, that fact is logged.

**Comment by mjordan** [comment]
@MarcusBarnes mind if I rename this manipulator to NormalizeDate and merge this into master? I'll provide tests.... :smile:

**Comment by MarcusBarnes** [comment]
@mjordan NormalizeDate it is!  Thank you very much. :+1:

**Comment by mjordan** [comment]
I'll wait to merge this one into master until branch issue-120.2 is merged into master, otherwise there will be conflicts in the Metadatamanipuator test file.

**Comment by mjordan** [comment]
OK, I've merged issue-120.2 into master. Will add some tests for the new Normalize Date manipulator later today.

**Comment by MarcusBarnes** [comment]
Thank you @mjordan :thumbsup:

**Comment by mjordan** [comment]
Merged into  master with 3c16dfd5fef87f4b67507252998c776d98e21bac with tests. I've updated the [wiki page](https://github.com/MarcusBarnes/mik/wiki/Metadata-manipulator:-NormalizeDate) for this manipulator.

**Comment by MarcusBarnes** [comment]
:thumbsup:
